src/server/pkg/storage/fileset/storage.go

```diff
@@ -316,74 +300,60 @@ func (s *Storage) compactSpec(ctx context.Context, fileSet string, compactedFile
 
 // Delete deletes a fileset.
 func (s *Storage) Delete(ctx context.Context, fileSet string) error {
-	fileSet = applyPrefix(fileSet)
-	return s.objC.Walk(ctx, fileSet, func(name string) error {
-		if err := s.objC.Delete(ctx, name); err != nil {
+	return s.store.Walk(ctx, fileSet, func(name string) error {
+		oid := filesetObjectID(name)
+		if err := s.store.Delete(ctx, name); err != nil {
```

Comment:
We can get this in a later PR, but we are going to want this to just be mark tombstone on the prefix (fileSet). The file set deleter and tracker objects we setup should handle the recursive deletion of sub file sets, then ultimately the chunks.